### PR TITLE
Prepare for 7.x release: Update version to 7.1.0, add ChangeLog

### PR DESCRIPTION
**Pull request by jychen7**
# Which issue does this PR close?

minor release for 7.1.0

 # Rationale for this change

# What changes are included in this PR?
`dev/release/update_change_log-all.sh` is updated to respect `release branch`, else it always compare difference from latest master

However, it still includes issues that are closed between the tags, so I have to manually fix it at 6d84c51034be3f13c62eb5e3a0f61211c1b0ec2d

# Are there any user-facing changes?

### CODE REVIEW

**Comment by jychen7** [comment]
@alamb do we think we still want to fix clippy for `maint-*` branch?

**Comment by alamb** [comment]
> 
@alamb do we think we still want to fix clippy for maint-* branch?

 I think it would be helpful. I can get a PR up for it

File: datafusion-cli/Cargo.toml [hunk 1]
```diff
@@ -31,7 +31,7 @@ rust-version = "1.58"
 clap = { version = "3", features = ["derive", "cargo"] }
 rustyline = "9.0"
 tokio = { version = "1.0", features = ["macros", "rt", "rt-multi-thread", "sync", "parking_lot"] }
-datafusion = { path = "../datafusion", version = "7.0.0" }
+datafusion = { path = "../datafusion", version = "7.1.0" }
 arrow = { version = "9.0.0" }
 ballista = { path = "../ballista/rust/client", version = "0.6.0" }
 env_logger = "0.9"
```
Comment (jackwener) [inline datafusion-cli/Cargo.toml:37]: To unify the style, I think we can add a new line?

**Comment by alamb** [comment]
Here is a PR to fix clippy in 7.x branch: https://github.com/apache/arrow-datafusion/pull/2191

File: datafusion-cli/Cargo.toml (hunk 1 above)
Comment (alamb) [reply datafusion-cli/Cargo.toml:37]: Either is fine with me -- I think some editors are configured to remove trailing newlines by default so perhaps that is what happened here?

File: datafusion/CHANGELOG.md [hunk 2]
```diff
@@ -19,6 +19,14 @@
 
 # Changelog
 
+## [7.1.0](https://github.com/apache/arrow-datafusion/tree/7.1.0) (2022-04-10)
+
+[Full Changelog](https://github.com/apache/arrow-datafusion/compare/7.0.0...7.1.0)
+
+**Fixed bugs:**
+
+- By default, use only 1000 rows to infer the schema [\#2159](https://github.com/apache/arrow-datafusion/pull/2159)
```
Comment (alamb) [inline datafusion/CHANGELOG.md:28]: 👍

File: dev/release/update_change_log-all.sh [hunk 3]
```diff
@@ -21,10 +21,11 @@
 set -e
 
 # Usage:
```
Comment (alamb) [inline dev/release/update_change_log-all.sh:23]:
```suggestion
# Usage:
# CHANGELOG_GITHUB_TOKEN=<TOKEN> ./update_change_log-all.sh <branch> <release_tag> <base_tag>
# Example:
```

**Review by alamb — APPROVED** [review]
LGTM -- thank you @jychen7

File: datafusion-cli/Cargo.toml (hunk 1 above)
Comment (jychen7) [reply datafusion-cli/Cargo.toml:37]: yes, some editor may remove trailing newlines. Either is fine with me too.

this file didn't have newline at the end, so I just let it be
https://github.com/apache/arrow-datafusion/blob/maint-7.x/datafusion-cli/Cargo.toml

**Comment by alamb** [comment]
Thanks @jychen7  -- shall I start the process to create 7.1.0 ?

**Comment by jychen7** [comment]
@alamb yes, I think it is good to go with 7.1.0. Thanks

**Comment by alamb** [comment]
Cool -- thanks -- I will prepare a release for a vote - likely tomorrow

**Pull request merged**